packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -38,25 +41,122 @@ def ensure_routable(cmd, service_points):
+        app['container']['docker']['portMappings'] = [{
+            'hostPort': 0,
+            'containerPort': 9080,
+            'protocol': 'tcp',
+            'name': 'test',
+            'labels': {}
+        }]
+        if vip is not None:
+            app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
+        if network == 'USER':
+            app['ipAddress'] = {'networkName': 'dcos'}
+    app['constraints'] = [['hostname', 'CLUSTER', host]]
+    return app, uuid
+
+
+def mesos_vip_app(num, network, host, vip, ucr=False):
+    app = None
+    uuid = None
+    port = backend_port_st + num
+    if ucr is False:
+        app, uuid = get_test_app()
+    else:
+        app, uuid = get_test_app_in_ucr()
+    app['id'] = '/viptest/' + app['id']
+    app['mem'] = 16
+    app['cpu'] = 0.01
+    # define a health check that works with all the network options
+    app['healthChecks'] = [{
+        'protocol': 'MESOS_HTTP',
+        'path': '/ping',
+        'gracePeriodSeconds': 5,
+        'intervalSeconds': 10,
+        'timeoutSeconds': 10,
+        'maxConsecutiveFailures': 3,
+    }]
+    assert network != 'BRIDGE'
+    if network == 'HOST':
+        app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                     '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
+        app['portDefinitions'] = [{
+            'protocol': 'tcp',
+            'port': 0
+        }]
+        if vip is not None:
+            app['portDefinitions'][0]['labels'] = {'VIP_0': vip}
+        app['healthChecks'][0]['portIndex'] = 0
+    if network == 'USER':
+        app['ipAddress'] = {
+            'discovery': {
+                'ports': [{
+                    'protocol': 'tcp',
+                    'name': 'test',
+                    'number': port,
+                }]
+            }
+        }
+        app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                     '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py {}'.format(port)
+        app['ipAddress']['networkName'] = 'dcos'
+        if vip is not None:
+            app['ipAddress']['discovery']['ports'][0]['labels'] = {'VIP_0': vip}
+        app['healthChecks'][0]['port'] = port
+        app['portDefinitions'] = []
+    app['constraints'] = [['hostname', 'CLUSTER', host]]
+    log.info('app: {}'.format(json.dumps(app)))
+    return app, uuid
+
+
+def vip_app(num, container, network, host, vip):
+    if container == 'UCR':
+        return mesos_vip_app(num, network, host, vip, ucr=True)
+    if container == 'DOCKER':
+        return docker_vip_app(network, host, vip)
+    assert container == 'NONE', 'unkown container option {}'.format(container)
+    return mesos_vip_app(num, network, host, vip, ucr=False)
+
+
 def vip_test(dcos_api_session, r):
     r.log('START')
-    agents = list(dcos_api_session.slaves)
+    agents = list(dcos_api_session.all_slaves)
```

Comment:
Nit: the intention of this line is to make a new list that is a copy of dcos_api_session.all_slaves so that it can be mutated; in such a case, it is helpful to do something like ```agents = copy.copy(dcos_api_session.all_slaves)``` to explicitly show that this is a pure copy and not a list type-cast